fuzzers/aflcc_no_orig_dic/fuzzer.py

```diff
@@ -0,0 +1,378 @@
+        os.environ['FUZZER_LIB'] += ' -L/ -lAflccMock -lpthread'
+
+    if is_benchmark('curl'):
+        shutil.copy('/libAflccMock.so', '/usr/lib/libAflccMock.so')
+
+    if is_benchmark('systemd'):
+        shutil.copy('/libAFL.a', '/usr/lib/libFuzzingEngine.a')
+        ld_flags = ['-lpthread']
+        utils.append_flags('LDFLAGS', ld_flags)
+
+
+def add_compilation_cflags():
+    """Add custom flags for certain benchmarks"""
+    if is_benchmark('openthread'):
+        openthread_flags = openthread_suppress_error_flags()
+        utils.append_flags('CFLAGS', openthread_flags)
+        utils.append_flags('CXXFLAGS', openthread_flags)
+
+    elif is_benchmark('php'):
+        php_flags = ['-D__builtin_cpu_supports\\(x\\)=0']
+        utils.append_flags('CFLAGS', php_flags)
+        utils.append_flags('CXXFLAGS', php_flags)
+
+    # For some benchmarks, we also tell the compiler
+    # to ignore unresolved symbols. This is useful when we cannot change
+    # the build process to add a shared library for linking
+    # (which contains mocked functions: libAflccMock.so).
+    # Note that some functions are only defined post-compilation
+    # during the LLVM passes.
+    elif is_benchmark('bloaty') or is_benchmark('openssl') or is_benchmark(
+            'systemd'):
+        unresolved_flags = ['-Wl,--warn-unresolved-symbols']
+        utils.append_flags('CFLAGS', unresolved_flags)
+        utils.append_flags('CXXFLAGS', unresolved_flags)
+
+    elif is_benchmark('curl'):
+        dl_flags = ['-ldl', '-lpsl']
+        utils.append_flags('CFLAGS', dl_flags)
+        utils.append_flags('CXXFLAGS', dl_flags)
+
+
+def add_post_compilation_lflags(ldflags_arr):
+    """Add additional linking flags for certain benchmarks"""
+    if is_benchmark('libjpeg'):
+        ldflags_arr += libjpeg_turbo_asm_object_files()
+    elif is_benchmark('php'):
+        ldflags_arr += ['-lresolv']
+    elif is_benchmark('curl'):
+        ldflags_arr += [
+            '-ldl', '-lpsl', '/src/openssl/libcrypto.a', '/src/openssl/libssl.a'
+        ]
+    elif is_benchmark('openssl'):
+        ldflags_arr += ['/src/openssl/libcrypto.a', '/src/openssl/libssl.a']
+    elif is_benchmark('systemd'):
+        shutil.copy(
+            os.path.join(os.environ['OUT'],
+                         'src/shared/libsystemd-shared-245.so'),
+            '/usr/lib/libsystemd-shared-245.so')
+        ldflags_arr += ['-lsystemd-shared-245']
+
+
+def prepare_fuzz_environment(input_corpus):
+    """Prepare run for some benchmarks"""
+    afl_fuzzer.prepare_fuzz_environment(input_corpus)
+
+    # OUT env variable does not exists, it seems.
+    if os.path.isfile('/out/src/shared/libsystemd-shared-245.so'):
+        shutil.copy('/out/src/shared/libsystemd-shared-245.so',
+                    '/usr/lib/libsystemd-shared-245.so')
+
+
+def prepare_build_environment():
+    """Set environment variables used to build benchmark."""
+    utils.set_no_sanitizer_compilation_flags()
```

Comment:
Which functions are different from aflcc, can we directly import the aflcc functions and call them, see example https://github.com/google/fuzzbench/blob/master/fuzzers/mopt/fuzzer.py#L21
if build and fuzz are same, can just call those, otherwise lets call common function for most things.

Reply:
Done.